tools/mo/openvino/tools/mo/ops/slice_like.py

```diff
@@ -29,7 +29,7 @@ def infer(node):
         input_value = node.in_port(0).data.get_value()
         shape_like = node.in_port(1).data.get_shape()
 
-        new_shape = np.copy(input_shape)
+        new_shape = shape_array(input_shape.copy())
```

Comment:
Reverted back into ` shape_array(input_shape.copy())` because if `input_shape` is static then `new_shape` will be a simple numpy array, and if `shape_like` is dynamic then we will try to write `dynamic_dimension` into it and end up with a error

https://github.com/openvinotoolkit/openvino/blob/6ab34578feea9af43df825a638fdcd9da24fb22e/tools/mo/openvino/tools/mo/ops/slice_like.py#L36

![image](https://user-images.githubusercontent.com/5703039/150410924-f72a261a-be68-4bb8-a26c-d4f82a6853bc.png)
This error was actually happening in real model for OOB scenario.

To prevent that I decorate with `shape_array` so that `new_shape` is always a masked numpy type that potentially can store dynamic values.